### ISSUE TITLE
Support discussion of Internet of Things (IoT)-related applications of schema.org

### ISSUE THREAD

**Issue by danbri** [schemaorg/schemaorg#1272]
* Nearby: there's an adhoc mailing list: https://groups.google.com/forum/#!forum/sdo-iot-sync
* Dedicated github repo: https://github.com/iot-schema-collab
* Slack: https://iot-schema-collab.slack.com/ (contact davidjanes@iotdb.org to join)
* [Custom Search Engine of related sites/repos](https://cse.google.com/cse/publicurl?cx=013484121852858951051:iccg_nifbxi) (contact danbri, Michael Koster, Darko Anicic, Max Senges to add/edit the included sites)

There have been a variety of conversations around schema.org-meets-IoT and they would benefit from having a home closer to the project as well as an integrated entry point (for which I suggest iot.schema.org).

Given 1.) that there is already a large (in fact the largest) W3C Community Group devoted to (the highly related idea of) "[Web of Things](https://www.w3.org/community/wot/)" ) 2.) this is a topic that relates strongly to parties beyond the W3C environment e.g. [IAB](https://www.iab.org/activities/workshops/iotsi/)/IETF, I have taken the liberty of setting up a simple ad-hoc public mailing list, [sdo-iot-sync@googlegroups.com](https://groups.google.com/forum/#!forum/sdo-iot-sync), rather than our usual practice of initiating a new W3C Community Group. I suggest that we accompany this with a simple entry point at iot.schema.org and will proceed in that direction unless anyone flags a concern. More focussed and tightly-scoped collaborative activities might happen in dedicated (e.g. community) groups at W3C or elsewhere, but for now there is no simple place for discussion of "schema.org meets the Internet of Things". I'd like to fix that ASAP.

IoT-related applications are incredibly diverse and are highly inter-related with other application themes such as health-lifesci, e-commerce, geospatial, datasets, sensors etc. Many existing schema.org schemas are also highly relevant to IoT applications. To make the most of this potential, and to provide a place where IoT-oriented schema proposals, extensions and implementations can be discovered, let's create a simple one page gateway via a "stub" hosted extension, iot.schema.org. It remains to be seen whether actually hosting schemas there, rather than by adding to the core vocabulary or externally, is the right path. Quite likely we may do a little of each. For now I am concerned primarily with surfacing ad-hoc conversations in a way that allows more visibility and wider collaboration amongst those enthused about semantic interoperability for the Internet of Things.

Copying schema.org steering group for info / advice / sanity check: @rvguha @shankarnat @chaals @vholland @scor @mfhepp.

**Comment by thadguidry** [comment]
@danbri let us know when the mailing list is up and running.  I plan to contribute highly in this Domain. Thanks.

**Comment by danbri** [comment]
Mailing list is up but give us a few days for "running". There's a short "getting started" doc in the works too that might help give things a little direction.

**Comment by vholland** [comment]
Joining the larger W3C discussion seems like the best approach. I'd hate for the groups to be building train tracks from opposite coasts and then fail to meet in the middle.

**Comment by mediaprophet** [comment]
the Web of Things [1] group seems like an appropriate place to make
enquiries.

[1] https://www.w3.org/WoT/

**Comment by danbri** [comment]
Regarding "surfacing discussions", amongst others I've had a very encouraging chat with W3C's Dave Raggett (@draggett) who leads the W3C work (community group + interest group + plans for a working group), and with Phil Archer (@philarcher1, also of W3C) regarding possible mechanisms for W3C REC-track work to normatively cite community-based schemas such as ours. I don't believe there is a 1:1 relationship between "schema.org meets IoT/WoT" and any other group; it is rather that various aspects of schemas and various aspects of IoT will intersect over time, hence the suggestion to use an "sdo-iot-sync" list.

**Comment by danbri** [comment]
@shankarnat @chaals @nicolastorzec @scor @mfhepp et al, my suggestion for next steps is that we include a 1-page iot.schema.org in our next release, and its role will be to channel discussions towards mailing list + github. We shouldn't rush into any specific schemas but it would be great to pull together some otherwise scattered conversations about IoT schemas and semantic interoperability.

**Comment by tmarshbing** [comment]
@danbri, your proposal and next steps make sense to me. Looking forward to watching this progress.

**Comment by danbri** [comment]
I have made some updates.

1. our "work in progress" branch for next schema.org release (sdo-callisto aka version 3.2) now has a stub IoT section. This can be (pre)viewed at http://iot.webschemas.org/ and if we proceed as proposed here will eventually constitute http://iot.schema.org/
(it links to this issue, the mailing list, and to a discussion document).

2. There's a discussion document towards getting started with IoT + schema.org in the repo and on the draft site, see http://iot.webschemas.org/docs/iot-gettingstarted.html for (somewhat simple) HTML. Original was [in google docs](https://docs.google.com/document/d/1mE2hN83lXzqXA-RQYMURJxIlEktcZkFB252BQCCnkKk/edit) and is probably the best place to refine specifics of that text or its successors. I suggest Github and email for bigger picture discussions.

3. I've sent [a kickoff mail](https://groups.google.com/forum/#!topic/sdo-iot-sync/YOTtNUNiJlM) to the sdo-iot-sync@googlegroups.com mailing list mentioned above, and cc:'d our steering group. Please join the group; for now I have set it to being postable by the world but we'll want to restrict posts to members only soon for better spam filtering.

/cc @chaals @nicolastorzec @vholland @mfhepp @rvguha @tmarshbing @shankarnat @scor

**Comment by scor** [comment]
👍

**Comment by mfhepp** [comment]
+1

**Comment by danbri** [comment]
ping @chaals

**Comment by thadguidry** [comment]
@danbri [Emailed our mailing list to start on a TODO list](https://groups.google.com/forum/#!topic/sdo-iot-sync/ZUa9_9aYIHA) (along with a quick primer on middle-layers and players)

**Comment by rvguha** [comment]
This sounds like a great idea.

guha

**Comment by thadguidry** [comment]
Add #1286 as a TODO for this issue.  It will be useful and is needed for IOT vendors and publishers and subscribers.

**Comment by danbri** [comment]
* http://ieeexplore.ieee.org/document/7471373/?reload=true "Semantic Technologies for the IoT - An Inter-IoT Perspective" 

> The Inter-IoT project is aiming at the design and implementation of, and experimentation with, an open cross-layer framework and associated methodology, to provide voluntary interoperability among heterogeneous Internet of Things (IoT) platforms. The project is initially driven by uses cases from two domains: (e/m) Health and transportation and logistics in a port environment. While the Inter-IoT will provide interoperability across the software stack, here, we focus our attention on the semantic interoperability. In this context, we present a concise overview of existing IoT-related semantic approaches, which might either be directly applicable to, or serve as a source of inspiration for, the Inter-IoT applications.

I can't see a free-to-read copy of the article, sadly. But http://plasma.dimes.unical.it/events/I4T2016/PDF/PawlowskiIoTSemantics.pdf seems to be a related presentation.

**Comment by thadguidry** [comment]
3GPP standards already has 1-5 year view of IoT with use cases and MIoT Massive Internet of Things. (that's how we roll in the ICT industry) This is one of just many practical studies 
http://www.3gpp.org/ftp//Specs/archive/22_series/22.861/22861-e10.zip

I don't want to overload Schema.org with 3GPP but want to let folks know that Communications side of IoT is covered within it.  And I am here along with other Ericsson employees to help build bridges where/if need be.

IoT is not about Things or Devices, but rather its about Communications and Data Exchange...where Things and Devices happen to play a Role.  Luckily, our efforts here on Schema.org and IoT alignment are fairly easy compared to the humongous efforts needed within ICT and 3GPP to make it all work seamlessly without breaking a sweat for us on the Comm side.

**Comment by thadguidry** [comment]
SCADA is another similar forerunner to IoT dealing with Data and Interoperability.  Currently there are some ISO standards around SCADA communications.  We'll want at least 1 person from the SCADA community to help our plans. But I don't know of anyone myself.

**Comment by danbri** [comment]
https://twitter.com/godanSec/status/802230005770096642 gives a good glimpse of the real scope of IoT - in this case as applied to agriculture. It is important that we don't take one central stereotype (the smart lightbulb) as proxying for the entire diverse set of IoT scenarios.

**Comment by danbri** [comment]
See https://swit.smartsdk.eu/news/ for papers and presentations from a recent semweb-meets-iot workshop.

**Comment by danbri** [comment]
Please take a look: http://w3c.github.io/sdw/ssn/#TheSOSAontology http://w3c.github.io/sdw/ssn/#TheSSNontology 

* Classes: | Actuation | Actuator | FeatureOfInterest | ObservableProperty | Observation | Platform | Procedure | Result | Sample | Sensor |
* Properties: | hasFeatureOfInterest | hasResult | hasSample | hasValue | hostedBy | hosts | invokedBy | invokes | isFeatureOfInterestOf | isObservedBy | isResultOf | isSampleOf | madeObservation | observedProperty | observes | phenomenonTime | resultTime | usedProcedure |

(this is part of the work being done in the W3C Spatial Data on the Web group...)

**Comment by chicco785** [comment]
Hi all,
discussing with the FIWARE guys working on https://github.com/Fiware/dataModels there is a certain interest to link existing json schema to schema.org (which of course implies adding the needed "semantics") - given that one of the next steps in NGSI is the introduction of support for JSON-LD.

The work they have done so far on data models is based on GSMA harmonized data models work: http://www.gsma.com/connectedliving/wp-content/uploads/2016/11/CLP.26-v1.0.pdf

The data models are used in different smart city scenarios.

If @danbri or anyonelse is interested in liaising, let me know.

**Comment by RichardWallis** [comment]
@chicco785 Thanks for sharing this.

Taking a quick (airport gate) scan of the GSMA [harmonized data models work](http://www.gsma.com/connectedliving/wp-content/uploads/2016/11/CLP.26-v1.0.pdf) document from a Schema.org point of view I have an immediate suggestion that I believe would improve the linking to Schema.org.

Following the Schema.org approach each entity type described in the Generic Entity Data Model section (eg. AgriParcel) could benefit from being a subtype of an appropriately named generic type (DataThing?), which contains all the properties (currently captured under the Generic Attributes section). That generic type itself being a subtype of Schema [Thing](http://schema.org/Thing).

The resultant inheritance hierarchy looking like this:
```Thing > DataThing > AgriParcel ``` or ```Thing > DataThing > AgriCrop``` etc.
The initial ramifications of such a move would be:
- The _id_ property used in all the generic attributes would be superseded by the [_identifier_](http://webschemas.org/identifier) property inherited from [Thing](http://webschemas.org/Thing) (here I am referencing the definition of Thing from the upcoming next release of Schema.org - currently visible at [webschemas.org](http://webschemas.org) )
- The generically useful properties such as _name_, _alternateName_, _description_, etc. would become available for all subtypes.
- The IoT relevant entity types referenced would all share a common supertype enabling consistant future maintenance and enhancement.

As in all serialisations of data supported by Schema.org (Microdata, RDFa, JSON-LD) require the the the Type to be stated (eg. ``` "@type": "AgriCrop",```) I question the need for the required _type_ property defined in the Generic Attributes definitions.

~Richard.

**Comment by danbri** [comment]
Interesting from David Janes @dpjanes: http://www.slideshare.net/dpjanes/alexa-toronto-open-data-civic-tech-toronto-presentation

**Comment by dpjanes** [comment]
Hey, that's me lol.

D.

**Comment by dpjanes** [comment]
If you just want to see the "schema.org normalized data", here you go
https://github.com/dpjanes/alexa-opendata/tree/master/dst

Nothing surprising here, but a fun project

**Comment by danbri** [comment]
Noting onem2m work, http://www.onem2m.org/technical/developers-corner/tools/onem2m-ontologies

**Comment by danbri** [comment]
Placeholder: discussed (c) statements in https://oneiota.org/revisions/2417 at Wishi workshop.

**Comment by danbri** [comment]
Nearby, https://github.com/openconnectivityfoundation

**Comment by danbri** [comment]
I've created a quick Google Custom Search engine configured with a bunch of URLs of IoT/WoT sites, repos etc. at https://cse.google.com/cse/publicurl?cx=013484121852858951051:iccg_nifbxi --- I've shared access with Michael Koster (who I'm sat next to at a [workshop](https://github.com/t2trg/2017-07-wishi/blob/master/AGENDA.md)) and others, we can easily add more sites. I don't see any easy way of sharing the current list of URLs but will figure something out. 

What does this mean?

Basically you can go to:

    https://cse.google.com/cse/publicurl?cx=013484121852858951051:iccg_nifbxi  

... and search for 'cbor', 'thermostat', 'json schema', 'carbon monoxide' or whatever and find work across a range of initiatives, projects, ecosystems.

**Comment by rvguha** [comment]
Nice! This should be useful!

guha

**Comment by danbri** [comment]
https://twitter.com/s3works/status/888682104938278912

https://arxiv.org/abs/1707.00112

"""A study of existing Ontologies in the IoT-domain
Garvita Bajaj, Rachit Agarwal, Pushpendra Singh, Nikolaos Georgantas, Valerie Issarny
(Submitted on 1 Jul 2017)
Several domains have adopted the increasing use of IoT-based devices to collect sensor data for generating abstractions and perceptions of the real world. This sensor data is multi-modal and heterogeneous in nature. This heterogeneity induces interoperability issues while developing cross-domain applications, thereby restricting the possibility of reusing sensor data to develop new applications. As a solution to this, semantic approaches have been proposed in the literature to tackle problems related to interoperability of sensor data. Several ontologies have been proposed to handle different aspects of IoT-based sensor data collection, ranging from discovering the IoT sensors for data collection to applying reasoning on the collected sensor data for drawing inferences. In this paper, we survey these existing semantic ontologies to provide an overview of the recent developments in this field. We highlight the fundamental ontological concepts (e.g., sensor-capabilities and context-awareness) required for an IoT-based application, and survey the existing ontologies which include these concepts. Based on our study, we also identify the shortcomings of currently available ontologies, which serves as a stepping stone to state the need for a common unified ontology for the IoT domain.
Comments:	Submitted to Elsevier JWS SI on Web semantics for the Internet/Web of Things
Subjects:	Artificial Intelligence (cs.AI)
Cite as:	arXiv:1707.00112 [cs.AI]
 	(or arXiv:1707.00112v1 [cs.AI] for this version)"""

**Comment by jaimejim** [comment]
Hi Dan, you might want to add the following URLs to the search: 

https://github.com/IPSO-Alliance/*
https://github.com/core-wg/*
https://tools.ietf.org/wg/core/*

**Comment by bobvanluijt** [comment]
Hi all!

I'm also working on Semantic IoT concepts. I've also started a Semantic Internet of Things platform on Github.

Blogpost with the concept: https://bob.wtf/semantic-internet-of-things-42811e1ca7a7
Github repo with Semantic IoT project: https://github.com/weaviate/weaviate

Looking forward to your feedback :-)

**Comment by gyrard** [comment]
IoT ontology catalogues would be relevant to build IoT schema.org.

- Linked Open Vocabulary for Internet of Things (LOV4IoT)
[http://sensormeasurement.appspot.com/?p=ontologies](http://sensormeasurement.appspot.com/?p=ontologies)
LOV4IoT project: [http://lov4iot.appspot.com/](http://lov4iot.appspot.com/)

- Ontology catalogue for smart cities (Ready4SmartCities)
[http://smartcity.linkeddata.es/](http://smartcity.linkeddata.es/)

- Linked Open Vocabulary (LOV)
[http://lov.okfn.org/dataset/lov/](http://lov.okfn.org/dataset/lov/)

- Ontology catalogue for smart cities (OpenSesingCity)
[http://opensensingcity.emse.fr/scans/ontologies](http://opensensingcity.emse.fr/scans/ontologies)

**Comment by ragarwa2** [comment]
The following paper "4W1H in IoT semantics" might be useful.. https://ieeexplore.ieee.org/document/8509571
""
IoT systems are now being deployed worldwide to sense phenomena of interest. The existing IoT systems are often independent which limits the use of sensor data to only one application. Semantic solutions have been proposed to support reuse of sensor data across IoT systems and applications. This allows integration of IoT systems for increased productivity by solving challenges associated with their interoperability and heterogeneity. Several ontologies have been proposed to handle different aspects of sensor data collection in IoT systems, ranging from sensor discovery to applying reasoning on collected sensor data for drawing inferences. In this paper, we study and categorize the existing ontologies based on the fundamental ontological concepts (e.g., sensors, context, location, and more) required for annotating different aspects of data collection and data access in an IoT application. We identify these fundamental concepts by answering the 4Ws (What, When, Who, Where) and 1H (How) identified using the 4W1H methodology.
""

**Comment by ragarwa2** [comment]
Recently privacy aspects wrt IoT have also been looked into.. some useful links could be:

"GDPR-inspired IoT Ontology enabling Semantic Interoperability, Federation of Deployments and Privacy-Preserving Applications" (https://arxiv.org/pdf/2012.10314.pdf)

*Abstract*

> Testing and experimentation are crucial for promoting innovation and building systems that can evolve to meet high levels of service quality. IoT data that belong to users and from which their personal information can be inferred are frequently shared in the background of IoT systems with third parties for experimentation and building quality services. This data sharing raises privacy concerns especially since in most cases the data are gathered and shared without the user's knowledge or explicit consent or for different purposes than the one for which the data were initially gathered. With the introduction of GDPR, IoT systems and experimentation platforms that federate data from different deployments, testbeds and data providers must be privacy-preserving. The wide adoption of IoT applications in scenarios ranging from smart cities to Industry 4.0 has raised concerns with respect to the privacy of users' data collected using IoT devices. Many experimental smart city applications are also using crowdsourcing data. Inspired by the GDPR requirements, we propose an IoT ontology built using available standards that enhances privacy, enables semantic interoperability between IoT deployments and supports the development of privacy-preserving experimental IoT applications. On top, we propose recommendations on how to efficiently use the ontology within IoT testbed and federating platforms. Our ontology is validated for different quality assessment criteria using standard validation tools. We focus on "experimentation" without loss of generality, because it covers scenarios from both research and industry, that are directly linked with innovation.

**Comment by rektide** [comment]
Just found out about Google's [Digital Buildings repo](https://github.com/google/digitalbuildings)/work. Definitely significant overlap, pretty active. Here's a link to their website's ["model" page](https://google.github.io/digitalbuildings/ontology/docs/model).
